src/ReactiveSocket.cpp

```diff
@@ -75,6 +76,20 @@ void ReactiveSocket::requestSubscription(
   automaton->start();
 }
 
+void ReactiveSocket::requestFireAndForget(
```

Comment:
It's one thing to not be compatible in the internals. It's another thing to not be compatible from external API point of view. Looking at the Java unit test I linked to, doing something different in C++ implementation would mean that the method named the same (reactiveSocket.fireAndForget()) behaves differently across languages (in Java, one receives a subscription and needs to call request(), in C++ it's immediately dispatched).

To you and Yuri's comments about why FireAndForgetRequester exists - it's only to be on par with what we do in Java. If we make a decision that we want to dispatch immediately, then I agree that it's not necessary.